include/audit_events.hpp

```diff
@@ -123,86 +123,168 @@ inline bool wantAudit(const crow::Request& req)
         (req.method() == boost::beast::http::verb::put) ||
         (req.method() == boost::beast::http::verb::delete_) ||
         ((req.method() == boost::beast::http::verb::post) &&
-         checkPostAudit(req)))
+         !checkPostUser(req)))
     {
         return true;
     }
 
     return false;
 }
 
-inline void auditEvent(const char* opPath, const std::string& userName,
-                       const std::string& ipAddress, bool success)
+/**
+ * @brief Checks if request should include additional data
+ *
+ * - Accounts requests data may contain passwords
+ * - IBM Management console events data is not useful. It can be binary data or
+ *   contents of file.
+ * - User login and session data may contain passwords
+ *
+ * @return True if request's data should not be logged
+ */
+inline bool checkSkipDetail(const crow::Request& req)
 {
-    int code = __LINE__;
+    if (req.target().starts_with("/redfish/v1/AccountService/Accounts") ||
+        req.target().starts_with("/ibm/v1") ||
+        ((req.method() == boost::beast::http::verb::post) &&
+         checkPostUser(req)))
+    {
+        return true;
+    }
+    return false;
+}
 
-    char cnfgBuff[256];
-    size_t bufLeft = 256; // Amount left available in cnfgBuff
-    char* user = NULL;
-    size_t opPathLen;
-    size_t userLen = 0;
-    int rc;
-    int origErrno;
+/**
+ * @brief Checks if request's detail data should be logged
+ *
+ * @return True if request's detail data should be logged
+ */
+inline bool wantDetail(const crow::Request& req)
+{
+    switch (req.method())
+    {
+        case boost::beast::http::verb::patch:
+        case boost::beast::http::verb::post:
+            if (checkSkipDetail(req))
+            {
+                return false;
+            }
+            return true;
+
+        case boost::beast::http::verb::put:
+            return (!req.target().starts_with("/ibm/v1"));
+
+        case boost::beast::http::verb::delete_:
+            return true;
+
+        default:
+            // Shouldn't be here, don't log any data
+            BMCWEB_LOG_DEBUG << "Unexpected verb " << req.method();
+            return false;
+    }
+}
 
-    if (auditOpen() == false)
+inline void auditEvent(const crow::Request& req, const std::string& userName,
+                       bool success)
+{
+    if (!auditOpen())
     {
         return;
     }
 
-    opPathLen = std::strlen(opPath) + 1;
+    char cnfgBuff[256];
```

Comment:
It looks like - the code logic seems trying to combine 3 attrs (e.g. opPath, detail, user) into a limited-sized buffer, and each append-logic seems essentially same.

So, it could be refactored into a small append function and use it for each attr like

```
inline bool append_item_to_buffer(std::string& cnfgBuffer, size_t maxOutSize,
                                  const std::string& item)
{
    if (cnfgBuf.length() + item.length() > maxOutSize)
    {
        return false;
    }
    cnfgBuffer += item;
    return true;
}

inline void auditEvent(const crow::Request& req, const std::string& userName,
                       bool success)
{
    if (!auditOpen())
    {
        return;
    }

    constexpr size_t cnfgBuffMaxSize = 256;
    std::string cnfgBuffer;

    std::string opPath = "op=" + std::string(req.methodString()) + ":" +
                         std::string(req.target()) + " ";
    if (!append_item_to_buffer(cnfgBuffer, cnfgBuffMaxSize, opPath))
    {
        BMCWEB_LOG_WARNING << "Audit buffer too small, truncating:"
                           << " bufLeft=" << bufLeft
                           << " opPathLen=" << opPathLen;
    }

    if (wantDetail(req))
    {
        std::string detail = req.body + " ";

        if (!append_item_to_buffer(cnfgBuffer, cnfgBuffMaxSize, detail))
        {
            BMCWEB_LOG_WARNING << "Audit buffer too small for data:"
                               << " bufLeft=" << bufLeft
                               << " detailLen=" << detail.length();
        }
    }

    char* user = audit_encode_nv_string("acct", userName.c_str(), 0);
    if (user == NULL)
    {
        BMCWEB_LOG_WARNING << "Error encoding user for audit msg : " << errno;
    }
    else
    {
        // setup a unique_ptr to handle freeing memory from user
        std::unique_ptr<char, void (*)(char*)> userUP(user, [](char* ptr) {
            if (ptr != nullptr)
            {
                ::free(ptr);
            }
        });
        if (!append_item_to_buffer(cnfgBuffer, cnfgBuffMaxSize,
                                   std::string(user)))
        {
            // Username won't fit into fixed sized buffer. Leave it off.
            BMCWEB_LOG_WARNING << "Audit buffer too small for username:"
                               << " bufLeft=" << bufLeft
                               << " userLen=" << userLen;
        }
    }
....

```

Reply:
Thank you Myung, this is a good idea. I've changed to use this approach. Note: When adding opPath I do want to truncate it rather than just not include it. (So any audit message will have at least a partial Redfish path included.) So when opPath is added I'm using std::substr instead to get the truncation.